### PR TITLE
Add --use-application-binary to "flutter install"

### PR DESCRIPTION
**Pull request by jmagman**
Allow installation of pre-built binaries, including with flavors.

Tested by building an ad-hoc IPA of the [flavors example](https://github.com/flutter/flutter/tree/master/dev/integration_tests/flavors):
```
$ flutter build ipa --flavor free --export-method ad-hoc
Archiving dev.flutter.flavors.free...
Warning: Missing build name (CFBundleShortVersionString).
Warning: Missing build number (CFBundleVersion).
Action Required: You must set a build name and number in the pubspec.yaml file version field before submitting to the App Store.
Automatically signing iOS for device deployment using specified development team in Xcode project: S8QB4VV633
Running Xcode build...
 └─Compiling, linking and signing...                         4.0s
Xcode archive done.                                         17.4s
Built /Users/m/Projects/flutter/dev/integration_tests/flavors/build/ios/archive/Free App.xcarchive.

💪 Building with sound null safety 💪

Building ad-hoc IPA...                                              6.3s
Built IPA to /Users/m/Projects/flutter/dev/integration_tests/flavors/build/ios/ipa.
```
And 
```
$ flutter install --use-application-binary build/ios/ipa/Free\ App.ipa
Installing dev.flutter.flavors.free to iPad...
$ flutter install --use-application-binary build/ios/ipa/Free\ App.ipa --uninstall-only
Uninstalling dev.flutter.flavors.free from iPad...
```

https://github.com/flutter/flutter/issues/99607#issuecomment-1085164018

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/install.dart [hunk 1]
```diff
@@ -15,9 +16,18 @@ class InstallCommand extends FlutterCommand with DeviceBasedDevelopmentArtifacts
...
     usesDeviceTimeoutOption();
-    argParser.addFlag('uninstall-only',
-      help: 'Uninstall the app if already on the device. Skip install.',
-    );
+    argParser
+      ..addFlag(
+        'uninstall-only',
+        help: 'Uninstall the app if already on the device. Skip install.',
+      )
+      ..addOption(
```
Comment (christopherfujino) [inline packages/flutter_tools/lib/src/commands/install.dart:24]: can you make this a helper function in flutter_command.dart? https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L203

**Review by christopherfujino — APPROVED** [review]
LGTM with a nit about using a helper for the flag

**Review by christopherfujino — APPROVED** [review]
Still LGTM, thanks!

**Review by Jasguerrero — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
Google testing borked, merging.

**Pull request merged**